### DISCUSSION TITLE
Visualize diff between revisions

### DISCUSSION THREAD

**Discussion by h4x3rotab** [outline/outline#2167] [Ideas]
It's nice to see revisions are kept for future review. It could be even better if there's an user interface to check the diff between revisions visually. Here's an example from Google Docs:

![](https://i.pcmag.com/imagery/articles/01UUnsPYh2VThtlAupMet0T-2..1569470602.png)

An easy approach is to render the two revisions and feed them to a HTML diff tool (e.g. [htmldiff.js](https://github.com/inkling/htmldiff.js)), which works pretty well in highlight diffs by `<ins>` and `<del>` elements.

A clearer way could be running a diff on the doc source file and render the diff. This may result in many corner cases and need to handled with great care.

**Comment by KlausThornProgrammfabrik** [comment]
My company started to migrate from Trac to Outline and I used the diff feature in Trac's wiki quite often. Outline's email notifications of changes that my coworkers made are quite handy, but the link in those emails does not lead to changes but to the whole page. I would then have to read the whole page and try to remember what could be new or read the new and the old version and compare by eyeballing. Looking at a colored diff like in Trac wiki or Github (or Google Docs as shown above) would be a whole new level.

**Reply by almereyda** [reply]
Another idea to ease these kinds of workflows has been expressed in #3382 as a suggest mode.

**Reply by tommoor** [reply]
This is badly overdue, totally agree

**Reply by tommoor** [reply]
For anyone that finds this, the functionality was recently completed and will be included in v0.67.0 onwards.
